nni/experiment/config/utils/internal.py

```diff
@@ -198,3 +201,37 @@ def get_ipv4_address() -> str:
     addr = s.getsockname()[0]
     s.close()
     return addr
+
+def load_experiment_config(config_json) -> typing.Union[ExperimentConfig, RetiariiExeConfig]:
+    # avoid circular import
+    from nni.nas.experiment.config import RetiariiExeConfig
+    from ..experiment_config import ExperimentConfig
+    if 'experimentType' in config_json:
+        if config_json['experimentType'] == 'hpo':
+            return ExperimentConfig(**config_json)
+        elif config_json['experimentType'] == 'nas':
+            return RetiariiExeConfig(**config_json)
+        else:
+            raise KeyError(f'Unknown experiment_type: {config_json["experimentType"]}')
+    else:
+        # for backward compatibility, experiment config <= v2.8 does not have "experiment_type"
+        if 'executionEngine' in config_json:
+            return RetiariiExeConfig(**config_json)
+        else:
+            return ExperimentConfig(**config_json)
+
+def get_experiment_cls_using_config(config_json):
```

Comment:
Seems the previous function should invoke this.

Reply:
good point, updated